### PR TITLE
feature: add FormatWithDeep

### PR DESCRIPTION
**Pull request by vekaputra**
add FormatWithDeep function so developer can output error with certain deep, without specifying deep when wrapping errors

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,5 @@
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+.idea
```
Comment (win-t) [inline .gitignore]: kurang karakter enter di belakang

File: wrapper.go [hunk 2]
```diff
@@ -63,32 +63,14 @@ func Walk(err error, fn func(*Error) bool) bool {
...
-		for _, l := range wrappedErr.StackTrace {
-			buff.WriteString("- ")
-			buff.WriteString(l.String())
-			buff.WriteString("\n")
-		}
-		return true
-	}) {
-		return buff.String()
-	}
+	return format(err, DefaultDeep)
```
Comment (win-t) [inline wrapper.go]: mending pake `-1` dari pada `DefaultDeep`, soalnya bisa aja orang yng generate errornya pake deep yng beda

File: wrapper.go [hunk 3]
```diff
@@ -127,3 +109,36 @@ func InErrorChain(err error, data interface{}) bool {
...
+		if first {
+			first = false
+		} else {
+			buff.WriteString("Caused by ")
+		}
+		buff.WriteString("Error: ")
+		buff.WriteString(wrappedErr.Error())
+		buff.WriteString("\n")
+		for i, l := range wrappedErr.StackTrace {
+			if i == deep {
```
Comment (win-t) [inline wrapper.go]: karna kalo deep nya `-1` ini jadi always false

File: wrapper.go [hunk 4]
```diff
@@ -127,3 +109,36 @@ func InErrorChain(err error, data interface{}) bool {
...
+			buff.WriteString(l.String())
+			buff.WriteString("\n")
+		}
+		return true
+	}) {
+		return buff.String()
+	}
+
+	return "Error: " + err.Error()
+}
```
Comment (win-t) [inline wrapper.go]: sama kurang karakter enter juga

File: .gitignore (hunk 1 above)
Comment (vekaputra) [reply .gitignore]: done afd3e1897c8e73411948049be7141d3782b21066

File: wrapper.go (hunk 2 above)
Comment (vekaputra) [reply wrapper.go]: done afd3e1897c8e73411948049be7141d3782b21066

File: wrapper.go (hunk 3 above)
Comment (vekaputra) [reply wrapper.go]: done afd3e1897c8e73411948049be7141d3782b21066

File: wrapper.go (hunk 4 above)
Comment (vekaputra) [reply wrapper.go]: done afd3e1897c8e73411948049be7141d3782b21066

**Pull request merged**